Services/Api/Cart/Controller.js

```diff
@@ -0,0 +1,138 @@
+const models = require('../../../database/models/index');
+const getTotally = require('./Helper');
+
+module.exports = {
+  create: async (req, res) => {
+    if (!req.body) return res.status(200).send({ message: 'Empty data' });
+    req.body.user = req.user.id;
+    req.body.createdAt = new Date(Date.now());
+    await models.Order.create(req.body).then((order) => {
+      res.send({
+        data: order,
+      });
+    }).catch((err) => {
+      res.status(500).send({
+        message: err,
+      });
+    });
+  },
+  viewOne: async (req, res) => {
+    if (!req.params._id) return res.status(200).send({ message: 'Empty data' });
+    await models.Order.findById(req.params._id, {
+      include: [
+        {
+          model: models.User,
+          as: 'users',
+        },
+        {
+          model: models.OrderProduct,
+          as: 'product_order',
+          include: [{
+            model: models.Product,
+            as: 'product_order',
+          }],
+        },
+      ],
+    }).then((order) => {
```

Comment:
WAT! That's the point of using async/away here... I mean, you can omit the `then` calls and such, and treat the return values from promises as is, e.g.

```js
const fn = async
 () => {
  const x = await Promise.resolve(42);
  console.log(typeof x); // number
  console.log(x); // 42
};
```

Reply:
I have to change all the controllers to grpc, the current controllers are incorrect, but, thanks for the observation!